File: assertion/tests/php/question5/normal.php
// `mumble` is a function that accepts an argument as string and returns a
// string as its output.
function mumble($input) {
    $result = '';
    for ($i = 0; $i < strlen($input); $i++) {
        $result .= ucfirst(strtolower(str_repeat($input[$i], $i + 1)));
        if ($i < strlen($input) - 1) {
            $result .= "-";
        }
    }
    
    return $result;
}

File: assertion/tests/c/question5/normal.c
#include <stdio.h>
#include <strings.h>

// `mumble` is a function that accepts an argument as string and returns a
// string as its output.
char *mumble(char *input)
{
    int inputLen = strlen(input);
    int triNum = (inputLen * (inputLen + 1)) / 2;
    int resultLen = triNum + inputLen - 1;
    char *result = malloc(resultLen);

    int pos = 0;
    for (int i = 0; i < inputLen; i++)
    {
        for (int j = 0; j <= i; j++)
        {
            char c = input[i];
            result[pos] = j == 0 ? toupper(c) : tolower(c);
            pos++;
        }
        if (pos != resultLen)
        {
            result[pos] = '-';
            pos++;
        }
    }

    return result;
}